airflow/kubernetes/pod_generator.py

```diff
@@ -412,16 +415,13 @@ def deserialize_model_file(path: str) -> k8s.V1Pod:
         """
         :param path: Path to the file
         :return: a kubernetes.client.models.V1Pod
-
-        Unfortunately we need access to the private method
-        ``_ApiClient__deserialize_model`` from the kubernetes client.
-        This issue is tracked here; https://github.com/kubernetes-client/python/issues/977.
         """
         if os.path.exists(path):
             with open(path) as stream:
                 pod = yaml.safe_load(stream)
         else:
-            pod = yaml.safe_load(path)
+            pod = None
+            log.warning("Model file %s does not exist", path)
 
         return PodGenerator.deserialize_model_dict(pod)
```

Comment:
Need to change type annotation of this method as well.

Reply:
Hi @uranusjr !
Sorry for having missed it. Addressed it at https://github.com/apache/airflow/pull/26641/commits/f61a7e0eeda900bf1879e70c17dc95dc4fe98dc6

Thanks!